front/src/modules/simulationResult/components/ChartHelpers/enableInteractivity.tsx

```diff
@@ -0,0 +1,384 @@
+import * as d3 from 'd3';
+import { mapValues } from 'lodash';
+import { zoom as d3zoom } from 'd3-zoom';
+import { pointer } from 'd3-selection';
+
+import {
+  gridX,
+  gridY,
+  interpolateOnPosition,
+  interpolateOnTime,
+  getAxis,
+  isSpaceTimeChart,
+} from 'modules/simulationResult/components/ChartHelpers/ChartHelpers';
+import {
+  ChartAxes,
+  LIST_VALUES,
+} from 'modules/simulationResult/components/simulationResultsConsts';
+import drawGuideLines from 'modules/simulationResult/components/ChartHelpers/drawGuideLines';
+import { dateIsInRange } from 'utils/date';
+import type {
+  Chart,
+  ConsolidatedRouteAspect,
+  GradientPosition,
+  Position,
+  SimulationD3Scale,
+  SpeedPosition,
+  ConsolidatedSpeedTime,
+  ConsolidatedPositionSpeedTime,
+  SpeedSpaceChart,
+  SimulationTrain,
+  PositionSpeedTime,
+  PositionsSpeedTimes,
+} from 'reducers/osrdsimulation/types';
+import type { typedEntries } from 'utils/types';
+import type { SpaceCurvesSlopesData } from 'modules/simulationResult/components/SpaceCurvesSlopes';
+import type { AreaBlock, GevPreparedData } from '../SpeedSpaceChart/prepareData';
+
+export const displayGuide = (chart: Chart | undefined, opacity: number) => {
+  if (chart?.svg) {
+    chart.svg.selectAll('#vertical-line').style('opacity', opacity);
+    chart.svg.selectAll('#horizontal-line').style('opacity', opacity);
+    chart.svg.selectAll('.pointer').style('opacity', opacity);
+  } else {
+    console.warn('attempt to display guide with no whart.svg set, check order of impl.');
+  }
+};
+
+export const updatePointers = (
+  chart: Chart | undefined,
```

Comment:
Chart is not undefined every where this function is called
```suggestion
  chart: Chart,
```